File: web/src/app/shared/interfaces/connection-property.ts
import { Property } from '../models/property';
import { Connection } from '../models/connection';

export class ConnectionProperty {
  id: number;
  property: Property;
  value: string;
  connection: Connection;


  constructor(id: number, property: Property, value: string, connection: Connection) {
    this.id = id;
    this.property = property;
    this.value = value;
    this.connection = connection;
  }
}

File: api/src/main/java/se/bjornlunden/blappadminapi/model/blapp/ConnectionProperty.java
package se.bjornlunden.blappadminapi.model.blapp;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

/**
 * @author Patrik Holmkvist on 2020-06-08
 */
@Entity
@Table(name = "connections_properties")
@Getter
@Setter
public class ConnectionProperty {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "connection_id")
    private Connection connection;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "property_id")
    private Property property;

    private String value;
}
